cli/src/main/java/org/jboss/as/cli/handlers/FilenameTabCompleter.java

```diff
@@ -60,45 +65,62 @@ protected int getCandidates(String buffer, List<String> candidates) {
         return matchFiles(buffer, translated, entries, candidates);
     }
 
+    public static FilenameTabCompleter newCompleter(CommandContext ctx) {
+        return Util.isWindows() ? new WindowsFilenameTabCompleter(ctx)
+                : new DefaultFilenameTabCompleter(ctx);
+    }
+    /**
+     * Translate a path that has previously been unescaped and unquoted.
+     * That is called at command execution when the calue is retrieved prior to be
+     * used as ModelNode value.
+     * @param path The unquoted, unescaped path.
+     * @return A path with ~ and default dir expanded.
+     */
     public String translatePath(String path) {
-        final String translated;
+        String translated;
         // special character: ~ maps to the user's home directory
         if (path.startsWith("~" + File.separator)) {
             translated = System.getProperty("user.home") + path.substring(1);
-        } else if (path.startsWith("~")) {
-            translated = new File(System.getProperty("user.home"))
-                    .getParentFile().getAbsolutePath();
-        } else if (!startsWithRoot(path)) {
+        } else if (path.startsWith("~") && !Util.isWindows()) {// Makes sense only on Unix.
```

Comment:
I noticed that Powershell recognises the "~\" shortcut. With the CLI. on all platforms, one can use this shortcut.

The else case seems fine, if Windows and the path starts with ~xxx, then this path will be returned as is, completion will not propose anything because the path is invalid on this platform. If the path doesn't start with ~, check for Absolute, if not, complete it, finally returns the original path.